### PR TITLE
[SPARK-11281][SPARKR] Add tests covering the issue.

### PR DESCRIPTION
**Pull request by zero323**
The goal of this PR is to add tests covering the issue to ensure that is was resolved by [SPARK-11086](https://issues.apache.org/jira/browse/SPARK-11086).

### CODE REVIEW

File: R/pkg/inst/tests/test_sparkSQL.R [hunk 1]
```diff
@@ -294,9 +294,21 @@ test_that("create DataFrame with complex types", {
 test_that("create DataFrame from a data.frame with complex types", {
   ldf <- data.frame(row.names=1:2)
   ldf$a_list <- list(list(1, 2), list(3, 4))
+  ldf$an_envir <- c(as.environment(list(a=1, b=2)), as.environment(list(c=3)))
+
   sdf <- createDataFrame(sqlContext, ldf)
+  collected <- collect(sdf)
+
+  expect_equivalent(ldf[, 1, FALSE], collected[, 1, FALSE])
```
Comment (sun-rui) [inline R/pkg/inst/tests/test_sparkSQL.R]: how about
expect_identical(ldf[, 1], collected[, 1])?

File: R/pkg/inst/tests/test_sparkSQL.R [hunk 2]
```diff
@@ -294,9 +294,21 @@ test_that("create DataFrame with complex types", {
...
+
+  asSortedList <- function(e) {
+      tmp <- as.list(e)
+      tmp[sort(names(tmp))]
+  }
 
-  expect_equivalent(ldf, collect(sdf))
+  expect_equal(
```
Comment (sun-rui) [inline R/pkg/inst/tests/test_sparkSQL.R]: seems
expect_equivalent(ldf$an_envir, collected$an_envir) should be OK?
 asSortedList is not necessary?

Comment (zero323) [inline R/pkg/inst/tests/test_sparkSQL.R]: @sun-rui You're right. Both changes should work just fine.

File: R/pkg/inst/tests/test_sparkSQL.R [hunk 3]
```diff
@@ -294,9 +294,13 @@ test_that("create DataFrame with complex types", {
 test_that("create DataFrame from a data.frame with complex types", {
   ldf <- data.frame(row.names=1:2)
   ldf$a_list <- list(list(1, 2), list(3, 4))
+  ldf$an_envir <- c(as.environment(list(a=1, b=2)), as.environment(list(c=3)))
```
Comment (sun-rui) [inline R/pkg/inst/tests/test_sparkSQL.R]: style: a = 1, b = 2;  c = 3

**Comment by sun-rui** [comment]
LGTM except a minor comment on coding style

**Comment by shivaram** [comment]
LGTM. Thanks @zero323 @sun-rui -- Merging this

**Pull request closed without merging**